src/elf/sym.rs

```diff
@@ -224,15 +224,14 @@ macro_rules! elf_sym_std_impl {
                     let bind = st_bind(self.st_info);
                     let typ = st_type(self.st_info);
                     let vis = st_visibility(self.st_other);
-                    write!(f,
-                           "st_name: {} {} {} {} st_shndx: {} st_value: {:x} st_size: {}",
-                           self.st_name,
-                           bind_to_str(bind),
-                           type_to_str(typ),
-                           visibility_to_str(vis),
-                           self.st_shndx,
-                           self.st_value,
-                           self.st_size)
+                    f.debug_struct("Sym")
+                        .field("st_name", &format_args!("{} {} {} {}", self.st_name, bind_to_str(bind), type_to_str(typ), visibility_to_str(vis)))
```

Comment:
These extra values aren't part of the name. Since you're displaying `st_info` and `st_other` now too, it might be better to move these to the field they are derived from. Same for the other `Sym` below.